### PR TITLE
Fix rename_duplicate to not require duplicates

### LINKED ISSUE #7539: astropy.table.Table.add_columns crash if rename_duplicate=True when there are no duplicates
When adding columns to an astropy table with add_columns (or add_column), an error occurs if using rename_duplicate=True when there are in fact no duplicates. From the description of rename_duplicate in the documentation, it seems that this is a bug (I would hope so; it would be nice if the user doesn't have to know whether there are duplicates in order to set the rename_duplicate keyword accordingly).

Here is the error: 

> Traceback (most recent call last):
>  File "<stdin>", line 1, in <module>
>  File "/home/mikea/anaconda2/envs/astroconda/lib/python2.7/site-packages/astropy/table/table.py", line 1569, in add_column
>    self.add_columns([col], [index], name, rename_duplicate=rename_duplicate)
>  File "/home/mikea/anaconda2/envs/astroconda/lib/python2.7/site-packages/astropy/table/table.py", line 1704, in add_columns
>    existing_names.add(new_name)
>UnboundLocalError: local variable 'new_name' referenced before assignment

Here is some example code that produces the error:

    import numpy as np
    from astropy.table import Table, Column
    t = Table(data=[np.arange(6)], names='a')
    t.add_column(Column(data=np.random.random(6), name='c'), rename_duplicate=True)

### PR DESCRIPTION
**Pull request by Mikea1985**
Fix issue #7539, including test to check correct functionality.

EDIT: Fix #7539

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -325,6 +325,8 @@ astropy.table
   object was not releasing the memory due to a reference cycle
   in the column ``info`` attributes. [#6277, #7448]
 
+- Fixed a bug in add_columns (issue #7539) where ``rename_duplicate=True`` would cause an error if there were in fact no duplicates. [#7540]
```
Comment (bsipocz) [inline CHANGES.rst]: Could you please move the entry down to the 2.0.8 section?

Comment (taldcroft) [reply CHANGES.rst]: Also break the line at a reasonable point, matching the formatting of other changelog entries.

Comment (taldcroft) [reply CHANGES.rst]: And say
```
.. bug in ``add_columns`` method where ..
```
Finally, drop the `in fact`.  Best to keep the language simple and direct here.

Comment (Mikea1985) [reply CHANGES.rst]: Thanks for your comments. I have made the requested changes. 
May I ask, why did I need to put the changelog entry in the 2.0.8 section? I see zero other entries down there, but a whole bunch up in the 3.1 section (which was why I assumed that was where it belonged).

**Comment by bsipocz** [comment]
The failures are totally unrelated, so I'm merging this now.

Thank you @Mikea1985 for the contribution!

**Pull request merged**